src/pages/groups/types/groupDetails.ts

```diff
@@ -0,0 +1,11 @@
+import { GroupModel, GroupRoles } from '../../../api/model/group.model'
+import { UserModel } from '../../../api/model/user.model'
+
+export type GroupDetails = GroupModel & {
+  members: (UserModel & {
+    joinedAt: Date
+    role: GroupRoles
+  })[]
+  owner: UserModel
+  role: GroupRoles
```

Comment:
maybe this could be renamed to something like userRole or currentUserRole, so it's clearer which role it's referring to